web/packages/teleterm/src/ui/services/deepLinks/deepLinksService.ts

```diff
@@ -88,17 +89,57 @@ export class DeepLinksService {
         break;
       }
       case '/authenticate_web_device': {
-        // TODO (avatus): add authenticate device web confirmation dialog
-        // this case is a stub and will not be reached
+        await this.askAuthorizeDeviceTrust(result.url);
         break;
       }
     }
   }
 
+  /**
+   * askAuthorizeDeviceTrust opens a dialog asking the user if they'd like to authorize
+   * a web session with device trust. If confirmed, the web session will be upgraded and the
+   * user will be directed back to the web UI
+   */
+  private async askAuthorizeDeviceTrust(
+    url: AuthenticateWebDeviceDeepURL
+  ): Promise<void> {
+    const { id, token } = url.searchParams;
+
+    const result = await this.loginAndSetActiveWorkspace(url);
+    if (!result.isAtDesiredWorkspace) {
+      return;
+    }
+
+    const { rootClusterUri } = result;
+    const rootCluster = this.clustersService.findCluster(rootClusterUri);
+    if (!rootCluster) {
+      return;
+    }
```

Comment:
Nit, but this defensive check is not necessary. If we already know that we're in the right workspace, then by necessity the underlying cluster must exist.

Even if for some reason it was gone, it's probably better to let the code crash than silently return with no message. This would be okay if we expected `rootCluster` to be regularly missing while a deep link is called, but I don't think we do.

In the case of returning after `if (!result.isAtDesiredWorkspace)`, it's okay to return with no message because the user just closed a modal, so they totally expect that nothing is going to happen. If a cluster is missing at this point, it's an unexpected situation.

Reply:
Thanks, removed